### DISCUSSION TITLE
Seedphrases with web3auth supported?

### DISCUSSION THREAD

**Discussion by YZhenY** [Web3Auth/Web3Auth#411] [Help]
Hey I saw web3auth website that there is a web3 (seed phrase/connect wallet) option to connect users as well. 

How is this done? Is there a web3auth integration part or is it purely using ethers.js to connect to an existing seed phrase (ie doesn't need web3auth).

Cant find any example of this in the docs. Can someone explain to me technically how to set this up step by step?

—————

Moreover, I am currently using custom auth, but want to allow users the option to connect to their own wallet as well. 

I noticed that normal web3auth login flow (which includes own wallet connect) uses @web3auth/web3auth package while web3auth custom auth uses @web3auth/core + @web3auth/openlogin-adapter. 

Does this mean that I would need to maintain TWO separate login flows if I want both custom auth and own wallet connect? Is there a way for custom auth's packages to support own wallet connect as well?

For example do I need to do this (say I am using react's context for each login flow)?:

Context1.tsx:
import { Web3Auth } from "@web3auth/web3auth";
const web3auth = new Web3Auth({...});
const web3authProvider = await web3auth.connect();

AND

Context2.tsx
import { Web3AuthCore } from "@web3auth/core";
const web3auth = new Web3AuthCore({...});
const web3authProvider = await web3auth.connectTo();

**Comment by YZhenY** [comment]
"How is this done? Is there a web3auth integration part or is it purely using ethers.js to connect to an existing seed phrase (ie doesn't need web3auth)."

We have a seedphrase provider that you, but its not the most used at the moment @himanshuchawla009 can you share more. Using simply ethers.js as you suggest would just work as well

"Does this mean that I would need to maintain TWO separate login flows if I want both custom auth and own wallet connect? Is there a way for custom auth's packages to support own wallet connect as well?"

No.

The wallet-connect provider is bundled in the @web3auth/ethereum-provider package https://www.npmjs.com/package/@web3auth/ethereum-provider

**Reply by qqqllppp** [reply]
@YZhenY Wouldn't both using `ethers.js` or `@web3auth/ethereum-provider` both need me to define a secondary `provider`?

So my first provider would be a custom auth:
```
import { Web3AuthCore } from "@web3auth/core";
const web3auth = new Web3AuthCore({...});
const web3authProvider = await web3auth.connectTo();
```

My second provider for anyone who wants to use their own private keys / seed phrase (not web3auth's generated private keys) using `ethers.js` or `@web3auth/ethereum-provider`:
```
const provider = // get provider from Infura or Alchemy
// plugin ethers.js or @web3auth/ethereum-provider
```

So basically if I want my dapp to have both custom auth AND external wallet connect, I would need to maintain TWO login flow right?

**Reply by YZhenY** [reply]
Yes, this is accurate

**Comment by nickfujita** [comment]
Hello, I'm also looking for this feature. Seem it was also posted as an issue on this repo, but it didn't get a response:
https://github.com/Web3Auth/Web3Auth/issues/225

In the web3auth docs for [Connect External Wallets](https://web3auth.io/docs/connect-external-wallets), it shows the following:
![image](https://user-images.githubusercontent.com/6342442/185896991-49aa58b5-358b-4d8a-9445-800bed710ec4.png)

Would be great to allow users to input their own seedphrase / private key in the same web3auth/ui modal instead of having to show the user a separate UI before getting to the web3auth modal, prompting them to select web3auth or BYO keys UI flow.

**Reply by yashovardhan** [reply]
Hey Nick,

Sorry for not getting back to you promptly around this question. We are constantly developing new adapters. Somehow this particular adapter was not prioritized according to our community needs. We'll add this to our development cycle as soon as possible and update you accordingly. 

I'll highlight additional ways of implementing this directly on your earlier discussion thread.

**Comment by graphicmist** [comment]
Hi, I need help on working on custom wallet where user can enter their passphrase? How is this supported. Please provide few pointers. I am working on integrating with near chain

**Reply by yashovardhan** [reply]
Refer to this discussion thread https://github.com/Web3Auth/Web3Auth/discussions/551
